snapshots/native/native.go

```diff
@@ -135,7 +135,7 @@ func (o *snapshotter) Mounts(ctx context.Context, key string) ([]mount.Mount, er
 		return nil, err
 	}
 	s, err := storage.GetSnapshot(ctx, key)
-	t.Rollback()
+	defer t.Rollback()
```

Comment:
I don't think we should defer here. Everything we need is done by GetSnapshot. Just Rollback it.